src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1332,21 +1333,16 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
         ucs_debug("got error on DC flow-control endpoint, iface %p: %s", iface,
                   ucs_status_string(ep_status));
     } else {
-        status = ib_iface->ops->set_ep_failed(ib_iface, &ep->super.super,
-                                              ep_status);
-        if (status != UCS_OK) {
+        status  = ib_iface->ops->set_ep_failed(ib_iface, &ep->super.super,
+                                               ep_status);
+        log_lvl = uct_ib_iface_set_failure_log_level(status, ep_status, ib_iface);
+
+        if ((log_lvl == UCS_LOG_LEVEL_FATAL) || (ep_status != UCS_ERR_CANCELED)) {
```

Comment:
can we move `(ep_status != UCS_ERR_CANCELED)` to `uct_ib_iface_failure_log_level` as well?

Reply:
In this function, if the status is UCS_ERR_CANCELED, then we should not call uct_ib_mlx5_completion_with_err() at all, with neither log_level (since when this function is called with UCS_ERR_CANCELED, the arg (=ecqe) is null).
uct_ib_iface_failure_log_level() only sets the log_level, it doesn't call uct_ib_mlx5_completion_with_err().

Reply:
IMO better sperate fatal and non-fatal path

Reply:
why do we need to check `log_lvl == UCS_LOG_LEVEL_FATAL` and not only `ep_status != UCS_ERR_CANCELED`?

Reply:
what should happen if ep_status is UCS_ERR_CANCELED and the status returned from set_ep_failed is not UCS_OK?

Reply:
if status is canceled, uct_set_ep_failed always returns OK